src/vs/workbench/contrib/chat/browser/tools/textSearchTool.ts

```diff
@@ -107,6 +110,38 @@ export class TextSearchTool implements IToolImpl {
 		// Search for the text
 		const { results, messages } = await this._searchService.textSearch(query, _token);
 
+		// If we have a chat context, include references for each result
+		if (invocation.context) {
+			const model = this._chatService.getSession(invocation.context.sessionId) as ChatModel;
+			const request = model.getRequests().at(-1)!;
+
+			for (const result of results) {
+				const { resource, results } = result;
+				if (!results || results.length === 0) {
```

Comment:
Super nit: An empty array is already falsey so no need to also check its length. Also recommend not having two vars in scope with the name `results` just for clarity.

Reply:
thanks for this nit! I've updated the variable name

ah but an empty array is truthy since it's an object -- I updated the `if` statement to be a bit more concise, but we still need to check the length of results